lib/http_clients/creditas/person_api.ex

```diff
@@ -1,6 +1,67 @@
 defmodule HttpClients.Creditas.PersonApi do
   @moduledoc false
 
+  alias HttpClients.Creditas.PersonApi.{Address, Contact, MainDocument, Person}
+
+  @spec get_person_by_cpf(Tesla.Client.t(), String.t()) :: Person.t()
+  def get_person_by_cpf(client, cpf) do
+    query = "mainDocument.code=#{cpf}"
+
+    case Tesla.get(client, "/persons", query: query) do
+      {:ok, %Tesla.Env{status: 200, body: attrs}} ->
+        {:ok, build_person(attrs)}
+
+      {:ok, %Tesla.Env{} = response} ->
+        {:error, response}
+
+      {:error, reason} ->
+        {:error, reason}
+    end
+  end
+
+  defp build_person(attrs) do
+    %Person{
+      fullName: attrs["fullName"],
+      birthDate: attrs["birthDate"],
+      contacts: build_contacts(attrs),
+      addresses: build_addresses(attrs),
+      mainDocument: %MainDocument{
+        type: attrs["mainDocument"]["type"],
+        code: attrs["mainDocument"]["code"]
+      }
+    }
+  end
+
+  defp build_contacts(attrs) do
+    Enum.reduce(attrs["contacts"], [], fn contact, contacts ->
+      [
+        %Contact{
+          channel: contact["channel"],
+          code: contact["code"],
+          type: contact["type"]
+        }
+        | contacts
+      ]
+    end)
+  end
```

Comment:
será que não rola assim?
chama
`build_contacts(attrs["contacts"])`
```suggestion
  defp build_contacts(contacts) do
    Enum.map(contacts, fn contact ->
       struct(Contact, contact)
    end)
  end
```
nessa linha?

daí igualmente pra build_address

Reply:
`struct(Contact, contact)` dessa maneira acredito que não rola pois é ignorado
https://hexdocs.pm/elixir/Kernel.html#struct/2-examples

Reply:
faz assim entao:
defp build_contacts(contacts) do    Enum.map(contacts, fn contact -&gt;       struct(Contact, contact)    end)  end

      %Contact{
        channel: contact["channel"],
        code: contact["code"],
        type: contact["type"]
      }

Reply:
no caso
```suggestion
  defp build_contacts(contacts) do
    Enum.map(contacts, fn contact ->
      %Contact{
        channel: contact["channel"],
        code: contact["code"],
        type: contact["type"]
      }
    end)
  end
```

Reply:
menos complexo que um reduce e acumulando o resultado, a gnt só  quer um map do mapa pra struct

Reply:
boa, feito